src/subdomains/supporting/payment/services/transaction-helper.ts

```diff
@@ -434,6 +441,80 @@ export class TransactionHelper implements OnModuleInit {
     };
   }
 
+  async getInvoiceDetails(userDataId: number, txId: number): Promise<TxStatementDetails> {
+    return this.getTxStatementDetails(userDataId, txId, TxStatementType.INVOICE);
+  }
+
+  async getReceiptDetails(userDataId: number, txId: number): Promise<TxStatementDetails> {
+    return this.getTxStatementDetails(userDataId, txId, TxStatementType.RECEIPT);
+  }
+
+  private async getTxStatementDetails(
+    userDataId: number,
+    txId: number,
+    statementType: TxStatementType,
+  ): Promise<TxStatementDetails> {
+    const transaction = await this.transactionService.getTransactionById(txId, {
+      userData: true,
+      buyCrypto: { buy: true, cryptoRoute: true, cryptoInput: true },
+      buyFiat: { sell: true, cryptoInput: true },
+      refReward: { user: { userData: true } },
+    });
```

Comment:
We could call `getTxStatementDetails` directly or? Then we don't need the other two methods anymore